boundary/analyze.py

```diff
@@ -263,6 +263,21 @@ def func_class_arithmetics(fns):
     fns.tainted = (fns.border | fns.insider | fns.sidecar) & fns.in_vmlinux
     fns.und = (fns.sched_outsider - fns.outsider_opt) | fns.border | fns.sidecar
 
+def get_func_decl_strs(signatures, fmt):
+    """Generate function declaration strings. If both strong and weak
+    symbol exist, keep only one.
+    """
+    decl_strs = set()
+    for fn in signatures:
+        s = fmt.format(**decls[fn])
+        if s in decl_strs:
+            assert fn[0] in global_fn_dict.keys(), \
+                'Attempt to redirect a repeating local symbol'
```

Comment:
What does "repeating" mean?

Reply:
If you have two symbols with the same name, and imagine these two cases below,

1. the 1st you encountered is a global one, and the 2nd you encountered is a local one
2. the 1st you encountered is a local one, and the 2nd you encountered is a global one 

In these two cases, this algorithm behaves differently. Why is that?